frontend-react/src/hooks/network/UseRequestConfig.ts

```diff
@@ -0,0 +1,129 @@
+import { useCallback, useEffect, useMemo, useState } from "react";
+import axios, { AxiosPromise, Method } from "axios";
+
+import { RSRequestConfig } from "../../network/api/NewApi";
+import { SimpleError } from "../../utils/UsefulTypes";
+
+export interface RequestHookResponse<D> {
+    data: D | undefined;
+    loading: boolean;
+    error: string;
+    trigger: () => void;
+}
+
+const DataSentRequest = ["POST", "PATCH", "PUT"];
+export const needsData = (reqType: Method) =>
+    DataSentRequest.includes(reqType.toUpperCase());
+export const hasData = (req: RSRequestConfig): boolean =>
+    req.data !== undefined;
+export const deletesData = (reqType: Method) =>
+    reqType === "delete" || reqType === "DELETE";
+export const needsTrigger = (reqType: Method) =>
+    reqType !== "GET" && reqType !== "get";
+const typedAxiosCall = <D>(config: RSRequestConfig): AxiosPromise<D> => {
+    switch (config.method) {
+        case "POST":
+        case "post":
+            return axios.post<D>(config.url, config.data, config);
+        case "PUT":
+        case "put":
+            return axios.put<D>(config.url, config.data, config);
+        case "PATCH":
+        case "patch":
+            return axios.patch<D>(config.url, config.data, config);
+        case "DELETE":
+        case "delete":
+            return axios.delete<D>(config.url, config);
+        case "GET":
+        case "get":
+        default:
+            return axios.get<D>(config.url, config);
+    }
+};
+
+/*
+ *
+ * Generic is the type of data coming back from the server */
+const useRequestConfig = <D>(
+    config: RSRequestConfig | SimpleError
+): RequestHookResponse<D> => {
+    /* Boolean indicating if method needs to be triggered or not. */
+    const onlyCallOnTrigger = useMemo(() => {
+        if (config instanceof SimpleError) return true;
```

Comment:
Tried the approach of a dedicated boolean but this check has to precede any use of the return value of `createRequestConfig` to ensure that we aren't trying to access fields not present on the object. The abstracted boolean wasn't enough to satisfy this check, but I did refactor the `useEffect` so that it actually throws and catches the error properly in _that_ block. I removed the check from this, made it a `useCallback` function instead of `useMemo` value, and am passing in the valid config only so no check or throw or anything needed.

Reply:
I wrote up a whole thing, but it's probably easier to go through this on a call, since it's tied up in a number of the other issues I've brought up.